### ISSUE TITLE
Too many alignment tabs in matrix

### ISSUE THREAD

**Issue by siwelwerd** [StevenClontz/checkit#59]
By default, the matrix environment in the amsmath package only allows 10 alignment characters:

https://tex.stackexchange.com/questions/3519/how-to-use-more-than-10-tab-stops-in-bmatrix-or-other-amsmath-matrix-environment

CheckIt can inadvertently generate LaTeX code with more than this.  This can be fixed by the user by inserting e.g. `\setcounter{MaxMatrixCols}{20}` in the preamble.

I am not sure there is an elegant way for checkit to do anything; perhaps this could be detected and the user issued a warning though.  Additionally, documentation for LaTeX users (bless their hearts) should probably be written to specify which packages need to be included (I think just amsmath, amssymb, and enumerate)

**Comment by StevenClontz** [comment]
`\setcounter{MaxMatrixCols}{20}` is exactly what the LaTeX template exported to Overleaf by the CheckIt Viewer does.

LaTeX boilerplate is worse than you suggest: we wrap everything in our own custom LaTeX environments, so that header info should be mentioned somewhere in documentation.

**Comment by siwelwerd** [comment]
Yeah, I wasn't thinking about the Overleaf template, but a user (okay, it was me) that wants to get the latex snippets for their own purposes (in my case, assembling them into one giant PDF I can send to the copier to generate a reassessment carnival).

The need to define the environments is pretty obvious from a glance at the snippets, but I had (maybe naively) hoped that once I did that, things would just work.

**Comment by StevenClontz** [comment]
I also made a custom LaTeX document with checkit exercises recently: I'd take the document produced at https://checkit.clontz.org/demo/#/assessment and sub in/out the LaTeX sources you want (or produce several such documents and copy things in/out as needed).
